packages/shared/lib/core/profile-manager/api/events/newOutput.ts

```diff
@@ -7,23 +7,24 @@ import { Bech32Helper } from '@lib/bech32Helper'
 import { Converter } from '@lib/converter'
 import { get } from 'svelte/store'
 import { NewOutputEvent } from '../types/newOutputEvent'
+import { preprocessGroupedOutputs } from '@core/wallet/utils/outputs/preprocessGroupedOutputs'
 
-export async function handleNewOutputEvent(accountId: string, event: NewOutputEvent): Promise<void> {
+export function handleNewOutputEvent(accountId: string, event: NewOutputEvent): void {
     const account = get(activeAccounts)?.find((account) => account.id === accountId)
+    const output = event?.output
 
     const address =
-        event?.output?.address?.type === ADDRESS_TYPE_ED25519
-            ? Bech32Helper.toBech32(0, Converter.hexToBytes(event?.output?.address?.pubKeyHash?.substring(2)), 'rms')
+        output?.address?.type === ADDRESS_TYPE_ED25519
+            ? Bech32Helper.toBech32(0, Converter.hexToBytes(output?.address?.pubKeyHash?.substring(2)), 'rms')
             : ''
-    if (
-        event?.output?.address?.type === ADDRESS_TYPE_ED25519 &&
-        account?.depositAddress === address &&
-        !event?.output?.remainder
-    ) {
+    if (output?.address?.type === ADDRESS_TYPE_ED25519 && account?.depositAddress === address && !output?.remainder) {
         syncBalance(account.id)
-        addActivityToAccountActivitiesInAllAccountActivities(
-            account.id,
-            await new Activity().setFromOutputData(event?.output, account, event?.transactionInputs)
+
+        const processedOutput = preprocessGroupedOutputs(
```

Comment:
Can we assume that this output recieved is not grouped with another existing one?